internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -968,11 +969,29 @@ func modifyMappedVirtualDisk(
 		mountCtx, cancel := context.WithTimeout(ctx, time.Second*5)
 		defer cancel()
 		if mvd.MountPath != "" {
+			verityInfo := mvd.VerityInfo
 			if mvd.ReadOnly {
+				// The only time the policy is empty, and we want it to be empty
+				// is when no policy is provided, and we default to open door
+				// policy. In any other case, e.g. explicit open door or any
+				// other rego policy we would like to mount layers with verity.
```

Comment:
Why would we want to mount with verity if open door policy is explicitly set but not if the policy is empty (aka implicit open door policy)?

Reply:
in non-confidential containers scenario, we'll have an implicit open door policy enforced, but the policy will be empty.

Reply:
Why do we not want to read it if we have an implicit open door policy? Just for perf reasons?

Reply:
yes, for perf reasons.